controllers/templates/migrateconf.tmpl

```diff
@@ -0,0 +1,37 @@
+env {
+  name = atlas.env
+  {{- if .URL }}
+  url = "{{ .URL }}"
+  {{- end }}
+  migration {
+    {{- if .Migration.Dir }}
+    dir = "{{ .Migration.Dir }}"
+    {{- end }}
+    {{- if .Cloud.Token }}
+    dir = data.remote_dir.this.url
+    {{- end }}
+  }
+}
+
+{{- if .Cloud.Token }}
+atlas {
+  cloud {
+    {{- if .Cloud.URL }}
+    url = "{{ .Cloud.URL }}"
+    {{- end }}
+    token = "{{ .Cloud.Token }}"
+    {{- if .Cloud.Project }}
+    project = "{{ .Cloud.Project }}"
+    {{- end }}
+  }
+}
+{{- end }}
+
+{{- if .Data.RemoteDir.Name }}
+data "remote_dir" "this" {
+  name = "{{ .Data.RemoteDir.Name }}"
+  {{- if .Data.RemoteDir.Tag }}
+  tag = "{{ .Data.RemoteDir.Tag }}"
+  {{- end }}
+}
+{{- end }}
```

Comment:
sort blocks: atlas -> dir -> env. The env's URL is required value. No need to check here.
```suggestion
{{- if .Cloud.Token }}
atlas {
  cloud {
    token = "{{ .Cloud.Token }}"
    {{- if .Cloud.Project }}
    project = "{{ .Cloud.Project }}"
    {{- end }}
    {{- if .Cloud.URL }}
    url = "{{ .Cloud.URL }}"
    {{- end }}
  }
}
{{- end }}

{{- if .Data.RemoteDir.Name }}
data "remote_dir" "this" {
  name = "{{ .Data.RemoteDir.Name }}"
  {{- if .Data.RemoteDir.Tag }}
  tag = "{{ .Data.RemoteDir.Tag }}"
  {{- end }}
}
{{- end }}

env {
  name = atlas.env
  url  = "{{ .URL }}"
  migration {
    {{- if .Migration.Dir }}
    dir = "{{ .Migration.Dir }}"
    {{- end }}
    {{- if .Cloud.Token }}
    dir = data.remote_dir.this.url
    {{- end }}
  }
}
```

Reply:
Also, please add tests